### PR TITLE
Implement missing LangVersion checks for new features/relaxations introduced in VB14.

### LINKED ISSUE #5072: LangVersion option doesn't disallow some VB14 features
This is a port of DevDiv 1213953

### Readonly auto-properties
```
    Class TestClass

        Public Sub New()
            'Check assignment of readonly auto property
            Test = "Test"
        End Sub

        'Check readonly auto-properties
        Public ReadOnly Property Test As String
    End Class
```
### Region in method
```
    Class TestClass

        'Check region in method
        Public Sub RegionInMe()

#Region "This is a region in a method"
            Console.WriteLine("I'm in a region")
#End Region

        End Sub
    End Class
```

### CObj in attribute arguments
```
    Class TestClass

        'Check CObj in attribute arguments
        <System.ComponentModel.DefaultValue(CObj("Test"))>
        Public Property Test2 As String

    End Class
```

### Multiline strings
```
    Class TestClass

        'Check Multiline strings
        Dim test4 = "
        This is
        a muiltiline
        string"

    End Class
```

### Comments at the end of strings
```
    Class TestClass

        'Check comment handling
        Dim test5 As String = ""
        Dim chars = From c In test5 'This is a test of comments in a linq statement
                    Let asc = Asc(c) 'VS2015 can handle this
                    Select asc

    End Class
```

### TypeOf IsNot
```
    Class TestClass

        Sub Test()
        'Check TypeOf Isnot
        Dim test6 As String = ""
        If TypeOf test6 IsNot System.String Then Console.WriteLine("That string isn't a string")
        End Sub

    End Class
```

### Year-first date literals
```
    Class TestClass

        Sub Test()
        'Check year-first date literals
        Dim d = #2015-08-23#
        End Sub

    End Class
```
 
### Pragma
```
    Class TestClass

        Sub Test()
        'Check pragma
#Disable Warning BC42024
        Dim test7 As String 'Should have no "unused variable" warning
#Enable Warning BC42024
        End Sub

    End Class
```

### Partial interfaces and modules
```
Partial Module Module1
End Module

Partial Interface IFace
End Interface
```

### Read-write property can implement read-only interface
```
Interface IReadOnly
    ReadOnly Property Test As String
End Interface

Class ReadWrite
    Implements IReadOnly

    Public Property Test As String Implements IReadOnly.Test
End Class
```

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #5072.
Also fixed some issues around LangVersion handling, which were found along the way.

@gafter, @VSadov, @jaredpar, @agocke Please review.

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/Binding/Binder_Attributes.vb [hunk 1]
```diff
@@ -781,6 +781,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                     End If
                                     Return CreateErrorTypedConstant(node.Type)
                                 Else
+                                    If node.Syntax.Kind = SyntaxKind.PredefinedCastExpression Then
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Binding/Binder_Attributes.vb]: Why isn't this in the parser? It is a purely syntactic test, isn't it?

Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Binding/Binder_Attributes.vb]: Oh, no, I see it is dependent on the context being an attribute. Yes, I see why it belongs here.

File: src/Compilers/VisualBasic/Portable/Parser/BlockContexts/CompilationUnitContext.vb [hunk 2]
```diff
@@ -134,6 +138,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             Private _notClosedIfDirectives As HashSet(Of IfDirectiveTriviaSyntax) = Nothing
             Private _notClosedRegionDirectives As HashSet(Of RegionDirectiveTriviaSyntax) = Nothing
             Private _notClosedExternalSourceDirective As ExternalSourceDirectiveTriviaSyntax = Nothing
+            Private _regionsAreAllowedEverywhere As Boolean
+
+            Private _parser As Parser
+            Private _declrationBlocksBeingVisited As ArrayBuilder(Of VisualBasicSyntaxNode) ' CompilationUnitSyntax is treated as a declaration block for our purposes
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Parser/BlockContexts/CompilationUnitContext.vb]: declaration misspelled.

**Comment by gafter** [comment]
LGTM

**Comment by VSadov** [comment]
LGTM

File: src/Compilers/VisualBasic/Portable/Parser/BlockContexts/CompilationUnitContext.vb (hunk 2 above)
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Parser/BlockContexts/CompilationUnitContext.vb]: Thanks! I'll fix this.

**Pull request merged**